test-wasm.html

```diff
@@ -0,0 +1,107 @@
+<script src="wasm/harfbuzzjs.js"></script>
+<script>
+var utf8Decoder = new TextDecoder("utf8");
+function bufferToString(buffer) {
+  return utf8Decoder.decode(buffer.slice(0, buffer.indexOf(0)));
+}
+
+var utf8Encoder = new TextEncoder("utf8");
+
+function hb_tag(tag) {
+  return [...tag].reduce((x, y) => (x << 8) + y.codePointAt(0), 0);
+}
+
+Module().then(module => {
+  window.module = module;
+
+  fetch('test/subset/data/fonts/Roboto-Regular.ttf').then(res => res.arrayBuffer()).then(font => {
+    var buffer = module._malloc(font.byteLength);
+    module.HEAPU8.set(new Uint8Array(font), buffer);
+
+    var blob = module._hb_blob_create(buffer, font.byteLength, 0/*FIXME*/, 0, 0);
+    // module._hb_blob_get_length(blob), module._hb_face_count(blob)
+    var face = module._hb_face_create(blob, 0);
+    var font = module._hb_font_create(face);
+
+    var buffer = module._hb_buffer_create();
+    {
+      var text = utf8Encoder.encode('Your text to shape!');
```

Comment:
Since JS uses UTF-16 internally, why not use that instead of re-encoding the text.

Reply:
Just tried and didn't work for some reason, I don't see it as a performance bottleneck as JS-WASM is more of bottleneck itself but you are right it was nice to have.

Reply:
`Chrome 54 (ticket) and Opera 41, no other encoding types are available other than utf-8, in order to match the spec.` https://developer.mozilla.org/en-US/docs/Web/API/TextEncoder so JavaScript doesn't provide utf-16 even the fact it may use it itself.